datalad_dataverse/remote.py

```diff
@@ -152,6 +161,8 @@ def checkpresent(self, key):
             return False
 
     def transfer_store(self, key, local_file):
+        if not self.dataset_version == ':latest':
```

Comment:
This and line 205 are not working.

Reply:
Well it seems to work. Only `datalad push` is ignoring Exceptions?

Reply:
When no exception is raised `datalad push` actually works and a DRAFT is created on dataverse.

The log is excatly the same when it fails. It still says it succeded but nothing changed on dataverse:
```
publish(ok): . (dataset) [refs/heads/git-annex->origin:refs/heads/git-annex 62d6084..a5c49a4]                                                                                                 
publish(ok): . (dataset) [refs/heads/master->origin:refs/heads/master 424537b..37ed9ae]                                                                                                       
action summary:                                                                                                                                                                               
  publish (ok: 2)
```

Reply:
It seems to work when the files handled by annex are pushed
It does not work when it's a git remote. 

Steps to reproduce:
`git clone "datalad-annex::?type=external&externaltype=dataverse&encryption=none&url=http://localhost:8080/&doi=doi:10.5072/FK2/JSUZ6P" directory`
`cp filexy directory
``cd directory`
`datalad save && datalad push --to origin`

Reply:
I suppose that is misleading, because you pushed that before, right? Hence, git-annex knows the file is there already and doesn't even try to TRANSFER-STORE. Needs to be fresh and with no pushed content yet.

Reply:
To clarify: If you look and the results, it's only two: one for each branch pushed. No content pushed whatsoever, and no error reported from annex. Hence this push call didn't even try.

Reply:
Yeah there are no new files since only the datalad metadata gets uploaded. Which are just these two files which are changed when new files are saved into the datalad instance.
![Screenshot from 2022-06-18 14-48-19](https://user-images.githubusercontent.com/2464969/174441456-01b970b6-f22f-4a9a-81a0-19dbfda848b5.png)

Reply:
Ok, that doesn't look right indeed. I'll have a look at it on Monday!

Thanks a ton so far!